File: backend/src/main/java/com/awais/hr/module/contractor/service/ContractorService.java
package com.awais.hr.module.contractor.service;

import java.util.List;
import java.util.Map;

public interface ContractorService {
    List<Map<String, Object>> getContractors();
    void addContractor(Map<String, Object> body);
    List<Map<String, Object>> getAgreements(String contractorId);
    void addAgreement(Map<String, Object> body);
    List<Map<String, Object>> getTimesheets(String contractorId);
    void submitTimesheet(Map<String, Object> body);
    void actionTimesheet(String timesheetId, String status);
}

File: frontend/src/services/contractorService.js
import apiClient from './api';

export const getContractors = () => apiClient.get('/contractor/contractors');
export const addContractor = (dto) => apiClient.post('/contractor/contractors', dto);
export const getAgreements = (contractorId) => apiClient.get(`/contractor/contractors/${contractorId}/agreements`);
export const addAgreement = (dto) => apiClient.post('/contractor/agreements', dto);
export const getTimesheets = (contractorId) => apiClient.get(`/contractor/contractors/${contractorId}/timesheets`);
export const submitTimesheet = (dto) => apiClient.post('/contractor/timesheets', dto);
export const actionTimesheet = (timesheetId, status) => apiClient.post(`/contractor/timesheets/${timesheetId}/action`, { status });
